src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -136,11 +132,22 @@ private static async Task<WasmFetchResponse> CallFetch(HttpRequestMessage reques
             JSObject abortController = BrowserHttpInterop.CreateAbortController();
             CancellationTokenRegistration? abortRegistration = cancellationToken.Register(() =>
             {
+#if FEATURE_WASM_THREADS
+                if (!abortController.IsDisposed)
+                {
+                    abortController.SynchronizationContext.Send(static (JSObject _abortController) =>
+                    {
+                        BrowserHttpInterop.AbortRequest(_abortController);
+                        _abortController.Dispose();
+                    }, abortController);
+                }
+#else
                 if (!abortController.IsDisposed)
                 {
                     BrowserHttpInterop.AbortRequest(abortController);
+                    abortController.Dispose();
                 }
-                abortController.Dispose();
+#endif
```

Comment:
This should be valid

```suggestion
                if (!abortController.IsDisposed)
                {
#if FEATURE_WASM_THREADS
                    abortController.SynchronizationContext.Send(static (JSObject abortController) =>
                    {
#else
                        BrowserHttpInterop.AbortRequest(abortController);
                        abortController.Dispose();
#if FEATURE_WASM_THREADS
                    }, abortController);
#endif
                }
```

Reply:
Are you saying that I should shadow `abortController` name in the callback, so that I could have more DRY code ?

Reply:
We can further tweak it in follow-up PRs. I will merge it so that it has time to flow to Blazor over the weekend.